src/twister2/quarantine_plugin.py

```diff
@@ -95,37 +100,51 @@ def get_matched_quarantine(self, item: pytest.Item) -> QuarantineElement | None:
         """Return quarantine element if test is matched to quarantine rules"""
         scenario = item.originalname
         platform = get_item_platform(item)
-        architecture = item.config.twister_config.get_platform(platform).arch if platform else ''
+        if platform:
+            plat_spec = item.config.twister_config.get_platform(platform)
+            architecture = plat_spec.arch
+            simulation = plat_spec.simulation
+        else:
+            architecture = ''
+            simulation = ''
 
         for qelem in self.qlist:
             matched: bool = False
             if qelem.scenarios:
-                if scenario in qelem.scenarios:
-                    matched = True
-                else:
-                    matched = False
+                if (matched := _is_element_matched(scenario, qelem.scenarios, qelem.use_regex)) is False:
```

Comment:
this also can be in one line:
` if qelem.scenarios and (matched := _is_element_matched(scenario, qelem.scenarios, qelem.use_regex)) is False:`
(looks like quite long line)

Reply:
if line is too long you can try:
```
match = partial(_is_element_matched, user_regex=qelem.use_regex))
if qelem.scenarios and (matched := match(scenario, qelem.scenarios)):
```

Reply:
IMO: using partial here is too confusing. I combined those two ifs with breaking lines:
```
            if (qelem.scenarios
                    and (matched := _is_element_matched(scenario, qelem.scenarios)) is False):
                continue
```
flake8 is ok with that solution.